### PR TITLE
Update to new ClimaParams, CloudMicrophysics and Thermodynamics

### PR DESCRIPTION
**Pull request by trontrytel**
Most of the unit tests and experiments work. 

Some calibration tests are failing, but right now I don't understand where those missing parameters should be coming from.

### CODE REVIEW

File: src/K1DModel/parameters.jl [hunk 1]
```diff
@@ -1,25 +1,62 @@
...
+    "Assumed aerosol dry radius [m]"
     r_dry::FT
+    "Assumed aerosol standard deviation [-]"
     std_dry::FT
+    "Assumed aerosol hygroscopicity"
     κ::FT
 end
 
+function Kinematic1DParameters(td::CP.AbstractTOMLDict)
+    #name_map = (;
```
Comment (trontrytel) [inline src/K1DModel/parameters.jl:35]: @nefrathenrici - could you take a look at why the map didn't work here?

File: src/Common/parameters.jl [hunk 2]
```diff
@@ -1,21 +1,45 @@
...
+    "Switch to include precipitation formation terms"
+    precip_sources::Bool
+    "Switch to include precipitation sink terms"
+    precip_sinks::Bool
+    "Prescribed number concentration of cloud droplets (needed for some schemes) [1/m3]"
     prescribed_Nd::FT
 end
 
+function CommonParameters(td::CP.AbstractTOMLDict)
+    #name_map = (;
```
Comment (trontrytel) [inline src/Common/parameters.jl:27]: @nefrathenrici - could you take a look at why the map didn't work here?

File: test/Project.toml [hunk 3]
```diff
@@ -25,14 +26,11 @@ Thermodynamics = "b60c26fb-14c3-4610-9d3e-2d17fe7ff00c"
 UnPack = "3a884ed6-31ef-47d7-9d2a-63182c4928ed"
 
 [compat]
-CLIMAParameters = "0.7"
 ClimaCore = "0.11"
```
Comment (trontrytel) [inline test/Project.toml:29]: I will add those compats back

File: src/K1DModel/parameters.jl [hunk 4]
```diff
@@ -1,25 +1,62 @@
...
+    #parameters = CP.get_parameter_values(td, name_map, "Kinematic1D")
+    #FT = CP.float_type(td)
+    #return Kinematic1DParameters{FT}(; parameters)
+    (; data) = td
+    FT = CP.float_type(td)
+    return Kinematic1DParameters(
+        FT(data["prescribed_flow_w1"]["value"]),
+        FT(data["prescribed_flow_t1"]["value"]),
+        FT(data["surface_pressure"]["value"]),
+        Int(data["qtot_flux_correction_flag"]["value"]),
```
Comment (trontrytel) [inline src/K1DModel/parameters.jl:53]: @nefrathenrici  - Is there a better way to handle a bool?

File: src/K1DModel/parameters.jl (hunk 1 above)
Comment (nefrathenrici) [reply src/K1DModel/parameters.jl:35]: I think on line 46 you need to splat the parameters: `return Kinematic1DParameters{FT}(; parameters...)`

File: src/K1DModel/parameters.jl (hunk 4 above)
Comment (nefrathenrici) [reply src/K1DModel/parameters.jl:53]: Is qtot_flux_correction_flag meant to be a Bool or an Int?
 ClimaParams supports bool types, so reading it in using the API will parse it as a Julia Boolean if the TOML type = bool

File: src/Common/parameters.jl (hunk 2 above)
Comment (nefrathenrici) [reply src/Common/parameters.jl:27]: Should be same as above, the parameters need to be splatted

**Review by nefrathenrici — APPROVED** [review]

**Review by sajjadazimi — APPROVED** [review]

**Pull request merged**